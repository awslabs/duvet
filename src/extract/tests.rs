// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

use super::*;

macro_rules! snapshot_test {
    ($name:ident) => {
        #[test]
        fn $name() {
            let contents = include_str!(concat!(
                env!("CARGO_MANIFEST_DIR"),
                "/specs/",
                stringify!($name),
                ".txt"
            ));

            let spec = Format::Ietf.parse(contents).unwrap();
            let sections = extract_sections(&spec);

            let results: Vec<_> = sections
                .iter()
                .flat_map(|(section, features)| {
                    let id = &*section.id;
                    features.iter().map(move |feature| (id, feature))
                })
                .collect();

            insta::assert_debug_snapshot!(stringify!($name), results);
        }
    };
}

snapshot_test!(rfc9000);
snapshot_test!(rfc9001);
snapshot_test!(rfc9114);
