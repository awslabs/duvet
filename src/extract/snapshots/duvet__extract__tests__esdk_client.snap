---
source: src/extract/tests.rs
expression: results
---
[
    (
        "initialization",
        Feature {
            level: Must,
            quote: [
                "On client initialization,",
                "the caller MUST have the option to provide a:",
            ],
        },
    ),
    (
        "initialization",
        Feature {
            level: Must,
            quote: [
                "If no [commitment policy](#commitment-policy) is provided the default MUST be [REQUIRE_ENCRYPT_REQUIRE_DECRYPT](../framework/algorithm-suites.md#require_encrypt_require_decrypt).",
            ],
        },
    ),
    (
        "initialization",
        Feature {
            level: Must,
            quote: [
                "If no [maximum number of encrypted data keys](#maximum-number-of-encrypted-data-keys) is provided",
                "the default MUST result in no limit on the number of encrypted data keys (aside from the limit imposed by the [message format](../format/message-header.md)).",
            ],
        },
    ),
    (
        "initialization",
        Feature {
            level: Should,
            quote: [
                "Once a [commitment policy](#commitment-policy) has been set it SHOULD be immutable.",
            ],
        },
    ),
    (
        "commitment-policy",
        Feature {
            level: Must,
            quote: [
                "The AWS Encryption SDK MUST provide the following commitment policies:",
            ],
        },
    ),
    (
        "maximum-number-of-encrypted-data-keys",
        Feature {
            level: Must,
            quote: [
                "Callers MUST have a way to disable this limit.",
            ],
        },
    ),
    (
        "forbid-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- `03 78` MUST be the default algorithm suite",
            ],
        },
    ),
    (
        "forbid-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [encrypt](encrypt.md) MUST only support algorithm suites that have a [Key Commitment](../framework/algorithm-suites.md#algorithm-suites-encryption-key-derivation-settings) value of False",
            ],
        },
    ),
    (
        "forbid-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [decrypt](decrypt.md) MUST support all algorithm suites",
            ],
        },
    ),
    (
        "require-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- `05 78` MUST be the default algorithm suite",
            ],
        },
    ),
    (
        "require-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [encrypt](encrypt.md) MUST only support algorithm suites that have a [Key Commitment](../framework/algorithm-suites.md#algorithm-suites-encryption-key-derivation-settings) value of True",
            ],
        },
    ),
    (
        "require-encrypt-allow-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [decrypt](decrypt.md) MUST support all algorithm suites",
            ],
        },
    ),
    (
        "require-encrypt-require-decrypt",
        Feature {
            level: Must,
            quote: [
                "- `05 78` MUST be the default algorithm suite",
            ],
        },
    ),
    (
        "require-encrypt-require-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [encrypt](encrypt.md) MUST only support algorithm suites that have a [Key Commitment](../framework/algorithm-suites.md#algorithm-suites-encryption-key-derivation-settings) value of True",
            ],
        },
    ),
    (
        "require-encrypt-require-decrypt",
        Feature {
            level: Must,
            quote: [
                "- [decrypt](decrypt.md) MUST only support algorithm suites that have a [Key Commitment](../framework/algorithm-suites.md#algorithm-suites-encryption-key-derivation-settings) value of True",
            ],
        },
    ),
    (
        "encrypt",
        Feature {
            level: Must,
            quote: [
                "The AWS Encryption SDK Client MUST provide an [encrypt](./encrypt.md#input) function",
                "that adheres to [encrypt](./encrypt.md).",
            ],
        },
    ),
    (
        "decrypt",
        Feature {
            level: Must,
            quote: [
                "The AWS Encryption SDK Client MUST provide an [decrypt](./decrypt.md#input) function",
                "that adheres to [decrypt](./decrypt.md).",
            ],
        },
    ),
]
